apt/message.go

```diff
@@ -0,0 +1,258 @@
+package apt
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// AptMessage represents a single RFC822 Apt message.
+type AptMessage struct {
+	code        int
+	description string
+	fields      map[string][]string
```

Comment:
Seems some fields are common, like "Last-Modified". Does it make sense to have it as struct fields (like Code and Description)?

Reply:
they are only common to 2 message types, and i was only interested in having a generic AptMessage interface. code, description and fields are common to all apt messages (specifically to all rfc822 type messages, which i was surprised to find no existing golang package for)